scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/ChangeSchema.java

```diff
@@ -133,6 +133,17 @@ public DataType(CqlType cqlType, List<DataType> typeArguments, UdtType udtType,
                     "Unexpected value of UdtType for this CQL type: " + cqlType.name());
         }
 
+        /**
+         * Constructs a new data type based on a given data type,
+         * but with a different value of frozen.
+         *
+         * @param dataType the data type that the constructed data type will be based on.
+         * @param isFrozen whether the constructed type will be frozen.
+         */
+        public DataType(DataType dataType, boolean isFrozen) {
```

Comment:
Just a small suggestion for a future improvement, no need to fix it here: the meaning of the `isFrozen` argument might not be obvious when you look at the code which uses a constructor of `DataType`. Perhaps we could introduce an enum of some kind?